File: LeetCode/Easy/1290-Convert_Binary_Number_In_A_Linked_List_To_Integer/1290-ConvertBinaryNumberInALinkedListToInteger.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var getDecimalValue = function (head) {
    let num = 0;
    let curr = head;

    while (curr != null) {
        num = (num << 1) + curr.val;
        curr = curr.next;
    }

    return num;
};

File: LeetCode/Easy/1290-Convert_Binary_Number_In_A_Linked_List_To_Integer/1290-ConvertBinaryNumberInALinkedListToInteger.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: Optional[ListNode]) -> int:
        num = 0
        curr = head

        while curr != None:
            num = (num << 1) + curr.val
            curr = curr.next

        return num
